### PR TITLE
Update dependencies and ESLint configuration migration

### PR DESCRIPTION
**Pull request by paazmaya**
I still have this thing before creating any release...

Plenty of ESLint related changes, but there is also `jsinspect` to `0.8.0`, which might bring something else to be considered...
https://github.com/danielstjules/jsinspect/releases/tag/0.8.0
or maybe not.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = function(grunt) {
 
     eslint: {
       options: {
-        config: '.eslintrc'
+        config: '.eslintrc.json'
```
Comment (stefanjudis) [inline Gruntfile.js]: why this rename?!

It's quite common to have it without `json` or?

Comment (paazmaya) [inline Gruntfile.js]: Tried to find the specific blog post about it, but so far came across only the part here http://eslint.org/docs/user-guide/configuring#configuration-file-formats mentioning that 

> Deprecated - use .eslintrc, which can be either JSON or YAML.

Comment (paazmaya) [inline Gruntfile.js]: Anyhow the version `2.0.0` of ESLint deprecated the extensionless configuration file, hence added it.

Comment (stefanjudis) [inline Gruntfile.js]: :+1:

**Comment by stefanjudis** [comment]
LGTM. :+1: Wanna merge and release or shall I?

**Pull request merged**